docs/application/dotnet/guides/user-interface/nui/vectorgraphics/TransformationAndComposition.md

```diff
@@ -0,0 +1,193 @@
+
+```
+
+![TransformationAndComposition_transform](./media/vectorgraphics_transfom.png)
+ 
+
+
+## Composition
+
+All `Drawable` can apply composition for visual effects such as alpha blending, masking(alpha/alpha inverse) and clipping.
+
+### Alpha blending
+Alpha blending can be applied by adjusting the `Opacity` property of the `Drawable` object and the alpha values of `FillColor` and `StrokeColor`.<br>
+Below is an example of alpha blending in various cases.
+
+```cs
+//generate a drawable group
+DrawableGroup drawableGroup = new DrawableGroup();
+
+//generate a round rectangle
+Shape rect = new Shape();
+rect.AddRect(50.0f, 50.0f, 200.0f, 200.0f, 20.0f, 20.0f);
+rect.FillColor = Color.Red;
+
+//Add the rectangle into the drawable group
+drawableGroup.AddDrawable(rect);
+
+//generate a circle
+Shape circle = new Shape();
+circle.AddCircle(200.0f, 200.0f, 100.0f, 100.0f);
+circle.FillColor = Color.Green;
+
+//add the circle into the drawable group
+drawableGroup.AddDrawable(circle);
+
+//set opacity value
+drawableGroup.Opacity = 0.5f;
+
+//generate an ellipse
+Shape ellipse = new Shape();
+
+ellipse.AddCircle(180.0f, 100.0f, 120.0f, 60.0f);
+
+//set ellipse translucent color
+ellipse.FillColor = new Color(0.0f, 0.0f, 1.0f, 0.5f);
+
+//add the eliipse into the canvas view
+canvasView.AddDrawable(ellipse);
+
+//add the drawable group into the canvas view
+canvasView.AddDrawable(drawableGroup);
+
+
+```
+
+![TransformationAndComposition_alphablending](./media/vectorgraphics_alphablending.png)
+
+### Clipping
+Clipping is a composition based on path information.<br>
+The intersection with clip `Drawable` is determined and only the resulting pixels from own `Drawable` are rendered.<br>
+
+Below is an example of clipping.
+
+```cs
+//generate two round rectangles
+Shape rect = new Shape();
+rect.AddRect(50.0f, 50.0f, 150.0f, 150.0f, 30.0f, 30.0f);
+rect.AddRect(100.0f, 100.0f, 150.0f, 150.0f, 30.0f, 30.0f);
+rect.FillColor = Color.Red;
+
+//generate a picture
+Picture picture = new Picture();
+picture.Load( IMAGE_PATH );
+picture.SetSize(new Size2D(400, 400));
+
+//clipping picture to rect
+picture.ClipPath(rect);
+
+//add the rect into the canvas view
+canvasView.AddDrawable(picture);
+```
+![TransformationAndComposition_clipping](./media/vectorgraphics_clipping.png)
+
+### Masking
+
+### Masking
+As for the `MaskType`, `Alpha` Masking and `InverseAlpha` Masking are provided.<br>
+The pixels of the own drawable and the complement to the mask drawable's pixels are alpha or inverse alpha blended.<br>
+
+| MaskType       | Description                                                                            |
+| -------------- | -------------------------------------------------------------------------------------- |
+| `Alpha`        | Only the part of the own drawable, which intersects with the mask drawable is visible. |
+| `InverseAlpha` | Only the part of the own which is not covered by the mask is visible.                  |
```

Comment:
The sentence is difficult to understand, "Only the part of the own..." what, drawable?
```suggestion
| `InverseAlpha` | Only the part of the own which is not covered by the mask is visible.                  |
```